app/src/checkupdatesdialog.cpp

```diff
@@ -0,0 +1,193 @@
+}
+
+CheckUpdatesDialog::~CheckUpdatesDialog()
+{
+    mNetworkManager->deleteLater();
+}
+
+void CheckUpdatesDialog::startChecking()
+{   
+#ifdef NIGHTLY
+    nightlyBuildCheck();
+#else
+    regularBuildCheck();
+#endif
+}
+
+void CheckUpdatesDialog::regularBuildCheck()
+{
+    mNetworkManager = new QNetworkAccessManager(this);
+    QUrl url("https://www.pencil2d.org/version/");
+
+    QNetworkRequest req;
+    req.setUrl(url);
+    req.setRawHeader("User-Agent", "Mozilla Firefox");
+
+    mNetworkManager->get(req);
+    connect(mNetworkManager, &QNetworkAccessManager::finished, this, &CheckUpdatesDialog::networkRequestFinished);
+}
+
+void CheckUpdatesDialog::nightlyBuildCheck()
+{
+    mTitleLabel->setText(tr("<b>You are using a Pencil2D nightly build</b>"));
+    mDetailLabel->setText(tr("Please go %1 here %2 to check new nightly builds.")
+                          .arg("<a href=\"https://www.pencil2d.org/download/#nightlybuild\">").arg("</a>"));
+    mDetailLabel->setOpenExternalLinks(true);
+    mProgressBar->setRange(0, 1);
+    mProgressBar->setValue(1);
+    mProgressBar->setTextVisible(false);
+    mDownloadButton->setEnabled(false);
+}
+
+void CheckUpdatesDialog::networkErrorHappened()
+{
+    mTitleLabel->setText(tr("<b>An error occurred while checking for updates</b>"));
+    mDetailLabel->setText(tr("Please check your internet connection and try again later."));
+    mProgressBar->setRange(0, 1);
+    mProgressBar->setValue(1);
+    mProgressBar->setTextVisible(false);
+    mDownloadButton->setEnabled(false);
+}
+
+void CheckUpdatesDialog::networkRequestFinished(QNetworkReply* reply)
+{
+    OnScopeExit(reply->deleteLater());
+
+    QNetworkReply::NetworkError errorCode = reply->error();
+    if (errorCode != QNetworkReply::NoError)
+    {
+        // show error message
+        qDebug() << reply->errorString();
+        networkErrorHappened();
+        return;
+    }
+
+    auto latestVersionString = QString::fromUtf8(reply->readAll()).trimmed();
+    qDebug() << latestVersionString;
+
+    bool isNewVersionAvailable = compareVersion(APP_VERSION, latestVersionString);
+    if (isNewVersionAvailable)
+    {
+        mTitleLabel->setText(tr("<b>A new version of Pencil2D is available!</b>"));
+        mDetailLabel->setText(tr("Pencil2D %1 is now available -- you have %2. Would you like to download it?")
+                              .arg(latestVersionString)
+                              .arg(APP_VERSION));
+        mProgressBar->hide();
+        mDownloadButton->setEnabled(true);
+    }
+    else
+    {
+        mTitleLabel->setText(tr("<b>Pencil2D is up to date</b>"));
+        mDetailLabel->setText(tr("Version") + " " APP_VERSION);
+        mProgressBar->setRange(0, 1);
+        mProgressBar->setValue(1);
+        mProgressBar->setTextVisible(false);
+        mDownloadButton->setEnabled(false);
+    }
+}
+
+bool CheckUpdatesDialog::compareVersion(QString currentVersion, QString latestVersion)
```

Comment:
We can eliminate this entirely by using `QVersionNumber` which has string parsing and version comparison functions built in.